xtask/src/backend.rs

```diff
@@ -0,0 +1,43 @@
+use crate::{nomt::NomtDB, sov_db::SovDB, timer::Timer};
+
+type Value = Vec<u8>;
+type Key = Vec<u8>;
+
+#[derive(Debug, Clone, clap::ValueEnum)]
+pub enum Backend {
+    SovDB,
+    Nomt,
+}
+
+#[derive(Clone, Debug)]
+#[allow(unused)]
+pub enum Action {
+    /// Batch of writes into the storage,
+    /// A value of None means delete that key
+    Write { key: Key, value: Option<Value> },
+    /// Batch of reads
+    Read { key: Key },
+}
+
+/// Trait implemented by all backends who wants to be benchmarked.
+pub trait DB {
+    /// Apply the given actions to the storage, committing them
+    /// to the database at the end.
+    ///
+    /// The function also accepts an optional timer that will be used to measure
+    /// the relevant parts of the backend that effectively apply the actions.
+    fn apply_actions(&mut self, actions: Vec<Action>, timer: Option<&mut Timer>);
+}
```

Comment:
Seems like the trait is not really necessary here, doesn't give any benefit.

Reply:
Why? The point of the trait is to be able to execute different workloads on different backends. The trait makes it possible to loop over all specified backends and execute the workload on them without handling all backends separately

Reply:
Fair enough